### PR TITLE
Rename Server to TcpServer

### PR DESCRIPTION
**Pull request by clue**
This simple PR renames the `Server` class to `TcpServer`.

A follow-up PR will introduce a new `Server` class which acts as a facade for the existing server classes. This class allows to start a `TcpServer` as-is and will prepare the code for the upcoming Unix domain socket server (#25) and upcoming `SecureServer` with auto-generated certificates (#95). This will work very similar to how the `Connector` class is a facade for the client side connector classes.

In other words, the follow-up PR will restore compatibility with consumer code so this does in fact not result in a BC break for most users.

### CODE REVIEW

**Comment by clue** [comment]
Rebased on current master to fix failing tests :shipit:
